src/nodes/chunk_dispatch/chunk_dispatch.c

```diff
@@ -290,8 +362,40 @@ chunk_dispatch_exec(CustomScanState *node)
 	/* Switch to the executor's per-tuple memory context */
 	old = MemoryContextSwitchTo(GetPerTupleMemoryContext(estate));
 
+#if PG15_GE
+	Oid partition_col_idx = InvalidOid;
+	TupleTableSlot *newslot = NULL;
+	if (dispatch->dispatch_state->mtstate->operation == CMD_MERGE)
+	{
+		partition_col_idx =
+			get_partition_attrno_from_tuple(ht->space, slot, state->colnames_list, false);
+		/* Row returned by child node does not have paritition col */
+		if (!OidIsValid(partition_col_idx))
+		{
+			List *actionStates =
+				dispatch->dispatch_state->mtstate->resultRelInfo->ri_notMatchedMergeAction;
+			ListCell *l;
+			foreach (l, actionStates)
+			{
+				MergeActionState *action = (MergeActionState *) lfirst(l);
+				CmdType commandType = action->mas_action->commandType;
+				if (commandType == CMD_INSERT)
```

Comment:
We should only have CMD_INSERT here, because we don't insert the chunk dispatch node otherwise, right?

Reply:
ChunkDispatch node is inserted when INSERT action is specified in MERGE, however it does not mean that MERGE cannot have other actions like UPDATE/DELETE. Thus we check for commandType.